### PR TITLE
Hosting Bundle uninstall should uninstall underlying runtimes

### PR DESCRIPTION
**Pull request by wtgodbe**
Right now, if you install the hosting bundle on a clean machine, then uninstall it, it'll leave behind the dotnet & asp.net runtimes. It should uninstall them - we were missing an `uninstall` command. Will need to backport this to all servicing branches once it's working.

Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1663163&view=results

https://github.com/dotnet/aspnetcore/issues/40957

### CODE REVIEW

File: src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs
Comment (dougbu) [inline src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:15]: How were things working without moving this `<Fragment/>` up❔ If everything was fine, let's limit the change to just the `UninstallCommand` additions.

Comment (wtgodbe) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:15]: They weren't, the search wasn't firing. Both things were a bug

Comment (dougbu) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:15]: Ah 🆗

**Review by dougbu — APPROVED** [review]

**Comment by wtgodbe** [comment]
New test build https://dev.azure.com/dnceng/internal/_build/results?buildId=1665177&view=results

File: src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs
Comment (joeloff) [inline src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:27]: I would consider adding `/norestart`

**Review by joeloff — CHANGES_REQUESTED** [review]

File: src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs
Comment (dougbu) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:27]: `Permanent="yes"` sounds very intentional. What are we _losing_ by enabling uninstall❔

Comment (joeloff) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:27]: Nothing AFAIK. It was originally authored this way. This was likely to deal with things like SxS installs of VS2015 (which included ANCM for IIS). We later added ANCM for VS 2017 so it's possible this was related to avoiding removing the runtimes in SxS scenario's since they did not support bundle-to-bundle reference counts

**Comment by wtgodbe** [comment]
https://dev.azure.com/dnceng/internal/_build/results?buildId=1701428&view=results

File: src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs
Comment (dougbu) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:46]: To be clear, I signed off but defer to you -- @joeloff. We need you for the **real** approval for both your expertise and your request for changes.

Comment (dougbu) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:46]: @wtgodbe perhaps I'm looking at a cached page. The part of this conversation about `Permanent="yes"` has defnitely been resolved. Did you add `/noRestart`❔

Comment (wtgodbe) [reply src/Installers/Windows/WindowsHostingBundle/DotNetCore.wxs:46]: I haven't, I don't think it's necessary for this PR

**Review by joeloff — APPROVED** [review]

**Pull request merged**
